src/raftstore/coprocessor/region_snapshot.rs

```diff
@@ -148,24 +141,21 @@ pub struct RegionIterator<'a> {
     end_key: Vec<u8>,
 }
 
-fn adjust_upper_bound(upper_bound: Option<&[u8]>) -> Option<&[u8]> {
-    if let Some(k) = upper_bound {
-        if k.is_empty() { None } else { Some(k) }
-    } else {
-        None
-    }
+fn set_upper_bound(iter_opt: IterOption, region: &Region) -> IterOption {
+    let upper_bound = match iter_opt.upper_bound() {
+        Some(k) if !k.is_empty() => keys::data_key(k),
```

Comment:
why need to set upper bound below if the IterOption already has the upper bound?